### PR TITLE
helm: fix argocd hanging

### PR DESCRIPTION
**Pull request by jenshu**
# Description

Fixes a couple of issues causing install/upgrade to fail with ArgoCD:
- The `gloo-resource-migration`, `gloo-resource-rollout`, and `gloo-resource-cleanup` jobs previously had a hard-coded `ttlSecondsAfterFinished` of 0, meaning the jobs were cleaned up immediately after they finished. This was too quick for ArgoCD, which after the job ran would seemingly try to clean up a job that was no longer there, and would be stuck syncing forever. To fix this, made the `ttlSecondsAfterFinished` configurable and defaulted to 60.
- The `gloo-resource-migration` job is only meant to run `pre-upgrade`, but since ArgoCD [maps both pre-install and pre-upgrade hooks](https://argo-cd.readthedocs.io/en/stable/user-guide/helm/#helm-hooks) to the same PreSync phase, it runs the migration job on both installs and upgrades. On installs, there was an ordering issue where the Gloo Edge CRDs were not installed before the migration job runs and the job would fail. Since the job is meant to only run on upgrades, added a check to only run migration steps if the CRDs are already there.

# Checklist:

- [ ] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [ ] If I updated APIs (our protos) or helm values, I ran `make -B install-go-tools generated-code` to ensure there will be no code diff
- [ ] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [ ] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: install/helm/gloo/templates/5-resource-migration-job.yaml [hunk 1]
```diff
@@ -35,6 +35,11 @@ spec:
           - /bin/sh
           - -c
           - |
+            if ! kubectl get crd upstreams.gloo.solo.io; then
+              echo "Gloo Edge crds are not installed; skipping migration job"
```
Comment (ianmacclancy) [inline install/helm/gloo/templates/5-resource-migration-job.yaml:39]: Is resource migration only for upstreams? The printout reads as if there are no crds at all and it looks like we are only checking upstreams. Might be more clear to say no upstream crds installed.

Comment (jenshu) [reply install/helm/gloo/templates/5-resource-migration-job.yaml:39]: right now it works on gateways and upstreams (and in the future any other CRs that we add to the helm chart).

i figured that just checking one arbitrary CRD would be enough, as if it's not there then it probably means Gloo Edge is not installed yet

i can update the message to be a bit more clear though - how does this sound: `Could not find Upstream CRD. Gloo Edge may not be installed yet. Skipping migration job`

Comment (jenshu) [reply install/helm/gloo/templates/5-resource-migration-job.yaml]: updated in 7518bfb

**Review by ianmacclancy — APPROVED** [review]
LGTM - kicked off flaking tests.

**Pull request merged**